### PR TITLE
Added service and environment to mass deploy

### PR DESCRIPTION
**Pull request by kula1922**
Service and environment are required when using mass deploy form.

### CODE REVIEW

File: src/ralph/ui/forms/deployment.py [hunk 1]
```diff
@@ -180,6 +187,30 @@ def _validate_preboot(preboot, row_number):
         )
 
 
+def _validate_service(service_name, row_number):
+    try:
```
Comment (andrzej-jankowski) [inline src/ralph/ui/forms/deployment.py]: ``if not ServiceCatalog.objects.filter(name=service_name).exists(): raise forms.ValidationError()`` will be better.

File: src/ralph/ui/forms/deployment.py [hunk 2]
```diff
@@ -180,6 +187,30 @@ def _validate_preboot(preboot, row_number):
...
+        raise forms.ValidationError(
+            "Row %s: "
+            "Couldn't find service with name %s" % (
+                row_number, service_name
+            )
+        )
+
+
+def _validate_environment(environment_name, row_number):
+    try:
```
Comment (andrzej-jankowski) [inline src/ralph/ui/forms/deployment.py]: See my previous comment...

File: src/ralph/deployment/tests/test_util.py [hunk 3]
```diff
@@ -171,3 +185,57 @@ def test_change_ip_address_dhcp_entry_fail(self):
...
+
+
+class CreateDeploymentsTest(TestCase):
+
+    def setUp(self):
+        venture_role = VentureRoleFactory()
+        device_environment = DeviceEnvironmentFactory()
+        service = ServiceCatalogFactory()
+        self.mass_deployment = MassDeploymentFactory()
+        self.user = User.objects.create_user(
```
Comment (andrzej-jankowski) [inline src/ralph/deployment/tests/test_util.py]: Use factory

File: src/ralph/deployment/tests/test_util.py [hunk 4]
```diff
@@ -171,3 +185,57 @@ def test_change_ip_address_dhcp_entry_fail(self):
...
+            "service": service.name,
+            "device_environment": device_environment.name,
+            "asset_identity": None,
+        }
+
+    def test_when_everything_works_fine(self):
+        EthernetFactory(mac="000000000000")
+        util.create_deployments([self.data], self.user, self.mass_deployment)
+
+        self.assertEqual(Deployment.objects.all().count(), 1)
```
Comment (andrzej-jankowski) [inline src/ralph/deployment/tests/test_util.py]: ``Deployment.objects.count()`` will be better.

File: src/ralph/deployment/tests/test_util.py [hunk 5]
```diff
@@ -171,3 +185,57 @@ def test_change_ip_address_dhcp_entry_fail(self):
...
+
+    def test_when_device_does_not_exist(self):
```
Comment (andrzej-jankowski) [inline src/ralph/deployment/tests/test_util.py]: Are you sure that this test name is correct?

File: src/ralph/deployment/tests/test_util.py [hunk 6]
```diff
@@ -171,3 +185,57 @@ def test_change_ip_address_dhcp_entry_fail(self):
...
+            device_info=device_info,
+        )
+        self.data.update({"asset_identity": asset.barcode})
+        util.create_deployments([self.data], self.user, self.mass_deployment)
+
+        deployments = Deployment.objects.all()
+        self.assertEqual(deployments.count(), 1)
+        self.assertEqual(deployments[0].device, device)
+
+    def test_when_device_does_not_exist_and_there_is_no_asset_identity(self):
```
Comment (andrzej-jankowski) [inline src/ralph/deployment/tests/test_util.py]: ditto

File: src/ralph/deployment/tests/utils.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+from ralph.discovery.tests.util import DeviceFactory
+
+
+class DeploymentFactory(DjangoModelFactory):
+    FACTORY_FOR = Deployment
+
+    device = factory.SubFactory(DeviceFactory)
+    mac = "000000000000"
+    ip = ""
+
```
Comment (andrzej-jankowski) [inline src/ralph/deployment/tests/utils.py]: missing blank line

**Comment by andrzej-jankowski** [comment]
:ok:

File: src/ralph/ui/forms/deployment.py [hunk 8]
```diff
@@ -432,7 +464,7 @@ def clean_csv(self):
             _validate_management_ip(management_ip, row_number)
             try:
                 venture_role = VentureRole.objects.get(
-                    venture__symbol=cols[6].strip().upper(),
+                    venture__symbol=cols[6].strip(),  #
```
Comment (zefciu) [inline src/ralph/ui/forms/deployment.py]: What's with this hash?

**Comment by zefciu** [comment]
:ok: after changes

**Pull request merged**